tests/strategies/area_agents/test_future_engine.py

```diff
@@ -32,21 +34,24 @@
 
 
 @pytest.fixture(name="future_engine")
-def future_engine_fixture():
+def future_engine_fixture() -> Generator[FutureEngine, None, None]:
```

Comment:
Why is this typing needed? Isn't always a `FutureEngine` object yielded?

Reply:
Yes, but when we declare types we are obliged to define the return type, and not the yield type. Therefore for generators, the return type is the following https://docs.python.org/3/library/typing.html#typing.Generator , YieldType=FutureEngine, SendType=None (because we do not send anything to the generator), ReturnType=None (because we have no return in the generator function)

Reply:
Very interesting! Thanks a lot for explaining and the link!